### PR TITLE
CTS sensors fixes for AAOS14

### PR DESCRIPTION
**Pull request by RajaniRanjan**
CTS test cases were failing for accelerometer,gyroscope and magnetic
sensors in AAOS.
1.testSensorAdditionalInfo
2.testGyroscopeUncalibrated_50hz_batching
3.testGyroscopeUncalibrated_fastest_batching
4.testSensorOperations_accelerometerLimitedAxes
5.testSensorOperations_gyroscope
6.testSensorOperations_gyroscopeLimitedAxesUncalibrated
7.testSensorOperations_magneticField
8.testAccelerometerLimitedAxes_automotive
9.testAccelerometer_automotive

Test Done:
run cts -m CtsSensorTestCases

Tracked-On: OAM-118145
Signed-off-by: RajaniRanjan <rajani.ranjan@intel.com>

### CODE REVIEW

**Review by balajim001 — APPROVED** [review]
LGTM

File: sensors/aidl/Sensor.cpp [hunk 1]
```diff
@@ -38,6 +40,7 @@ Sensor::Sensor(ISensorsEventCallback* callback)
       mLastSampleTimeNs(0),
       mCallback(callback),
       mMode(OperationMode::NORMAL) {
+        setAdditionalInfoFrames();
```
Comment (vilasrk) [inline sensors/aidl/Sensor.cpp:43]: Indentation is not proper.

File: sensors/aidl/Sensor.cpp [hunk 2]
```diff
@@ -122,7 +125,7 @@ void Sensor::run() {
             });
         } else {
             timespec curTime;
-            clock_gettime(CLOCK_REALTIME, &curTime);
+            clock_gettime(CLOCK_BOOTTIME, &curTime);
```
Comment (vilasrk) [inline sensors/aidl/Sensor.cpp:128]: Add reason for changing it to Boot time in commit message

File: sensors/aidl/Sensor.cpp [hunk 3]
```diff
@@ -220,10 +254,17 @@ std::vector<Event> Sensor::readEvents() {
                 vec3 = {
                     .x = 0,
                     .y = 0,
-                    .z = -9.8,
+                    .z = 9.8,
                     .status = SensorStatus::ACCURACY_HIGH,
                     };
                 event.payload.set<EventPayload::Tag::vec3>(vec3);
+                events.push_back(event);
+                for (const auto& frame : mAdditionalInfoFrames) {
```
Comment (vilasrk) [inline sensors/aidl/Sensor.cpp:262]: We use the same code base for CIV as well, and additional info may not be needed for CIV.
Can we add this code in condition based on TARGET_PRODUCT?

File: sensors/aidl/Sensor.cpp [hunk 4]
```diff
@@ -233,10 +274,32 @@ std::vector<Event> Sensor::readEvents() {
                     .z = 50.0,
                     .status = SensorStatus::ACCURACY_HIGH,};
                 event.payload.set<EventPayload::Tag::vec3>(vec3);
+                events.push_back(event);
+                for (const auto& frame : mAdditionalInfoFrames) {
```
Comment (vilasrk) [inline sensors/aidl/Sensor.cpp:278]: We use the same code base for CIV as well, and additional info may not be needed for CIV.
Can we add this code in condition based on TARGET_PRODUCT?

File: sensors/aidl/Sensor.cpp [hunk 5]
```diff
@@ -360,6 +462,29 @@ std::vector<Event> OnChangeSensor::readEvents() {
     return outputEvents;
 }
 
+void Sensor::setAdditionalInfoFrames() {
+    // https://developer.android.com/reference/android/hardware/SensorAdditionalInfo#TYPE_SENSOR_PLACEMENT
+
```
Comment (vilasrk) [inline sensors/aidl/Sensor.cpp:467]: We use the same code base for CIV as well, and additional info may not be needed for CIV.
Can we add this code in condition based on TARGET_PRODUCT?

**Review by vilasrk — CHANGES_REQUESTED** [review]

File: sensors/aidl/Sensor.cpp (hunk 2 above)
Comment (RajaniRanjan) [reply sensors/aidl/Sensor.cpp:128]: google is recommending CLOCK_BOOTTIME to calculate current time as it keeps running during suspend.

File: sensors/aidl/Sensor.cpp (hunk 3 above)
Comment (RajaniRanjan) [reply sensors/aidl/Sensor.cpp:262]: yes this can be done to support target specific features.

File: sensors/aidl/Sensor.cpp (hunk 4 above)
Comment (RajaniRanjan) [reply sensors/aidl/Sensor.cpp:278]: yes this can be done to support target specific features.

File: sensors/aidl/Sensor.cpp (hunk 5 above)
Comment (RajaniRanjan) [reply sensors/aidl/Sensor.cpp:467]: yes this can be done to support target specific features.

**Review by vilasrk — APPROVED** [review]
LGTM

**Review by vilasrk — APPROVED** [review]
LGTM

**Review by balajim001 — APPROVED** [review]
LGTM

**Pull request merged**